### PR TITLE
Update the compatibility section to have correct versions

### PR DESCRIPTION
**Pull request by Hunterness**
Also add new (empty) table for additions/deprecations/removals in 4.4

### CODE REVIEW

File: cypher/cypher-docs/src/docs/dev/deprecations-additions-and-compatibility.asciidoc [hunk 1]
```diff
@@ -1043,7 +1052,7 @@ All supported versions of Cypher ran on the same Neo4j kernel.
 However, this changed in Neo4j 3.4 when the runtime was excluded from the compatibility layer.
 When you run, e.g. a `CYPHER 3.1` query in Neo4j 3.5, the query is planned with the 3.1 planner, but run with 3.5 runtime and kernel.
 The compatibility layer changed again in Neo4j 4.0 and it now includes only the Cypher language parser.
-When you run a `CYPHER 3.5` query, e.g., in Neo4j 4.3, Neo4j parses the older language features, but uses the 4.3 planner, runtime, and kernel to plan and run the query.
+When you run a `CYPHER 3.5` query, e.g., in Neo4j 4.4, Neo4j parses the older language features, but uses the 4.4 planner, runtime, and kernel to plan and run the query.
```
Comment (davidoliverSP2) [inline cypher/cypher-docs/src/docs/dev/deprecations-additions-and-compatibility.asciidoc:1055]: If we want this to always be the most up to date version, we can use the attributes `{neo4j-version}` for _major.minor_, or `{neo4j-version-exact}` for _major.minor.patch_.

**Review by davidoliverSP2 — APPROVED** [review]
Thanks @Hunterness 

One comment, that doesn't necessarily block this PR. I can raise a card on our wall to resolve this in earlier versions and cherry pick forward.

File: cypher/cypher-docs/src/docs/dev/deprecations-additions-and-compatibility.asciidoc (hunk 1 above)
Comment (Hunterness) [reply cypher/cypher-docs/src/docs/dev/deprecations-additions-and-compatibility.asciidoc:1055]: I asked about that on Martins pr in this section, https://github.com/neo4j/neo4j-documentation/pull/1123#discussion_r643756982, and we could just have kept it as 4.3 here but I decided to update it when I anyway poked around in it (I guess it says something else in 4.2 anyway)

**Review by Lojjs — APPROVED** [review]
Looks good to me 👍

**Pull request merged**